docs/learn/smart-contract-developers/create-nft-collection-with-lsp7-tokenId.md

```diff
@@ -0,0 +1,956 @@
+- the `isNonDivisible` of the LSP7 token
+- the `totalSupply` of the LSP7 token
+- the `receiver` of the initial tokens of the LSP7 token
+
+### Imports
+
+Let's start creating an `LSP8Token.sol` file in the `contracts` folder and import:
+
+- the [`LSP8IdentifiableDigitalAsset.sol`](https://github.com/lukso-network/lsp-smart-contracts.git) from which we will inherit the main functionalities of the LSP8
+- the [`_LSP4_METADATA_KEY`](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#lsp4metadata) constant
+- the LSP7 token contract we just created so that we can use it in our `mint(...)` function
+
+```typescript
+import { LSP8IdentifiableDigitalAsset } from '@lukso/lsp-smart-contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.sol';
+import { _LSP4_METADATA_KEY } from '@lukso/lsp-smart-contracts/contracts/LSP4DigitalAssetMetadata/LSP4Constants.sol';
+import { LSP7Token } from './LSP7Token.sol';
+```
+
+### Contract
+
+#### Constructor
+
+The constructor will be in charge of setting the following parameters:
+
+- `name`: the name of the LSP8 token
+- `symbol`: the symbol of the LSP8 token
+- `owner`: the owner of the LSP8 token that will be able change the contract metadata and mint new LSP7 tokens
+- [`lsp4TokenType`](../../standards/tokens/LSP7-Digital-Asset.md##lsp4tokentype) of the LSP7 token
+- `lsp8TokenIdFormat`: the [format of the tokenId](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#lsp8tokenidformat) of the LSP8 token
+- `lsp4MetadataURI`: the [LSP4 Metadata](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#lsp4metadata)
+
+```typescript
+contract LSP8Token is LSP8IdentifiableDigitalAsset {
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        address newOwner_,
+        uint256 lsp4TokenType_,
+        uint256 lsp8TokenIdFormat_,
+        bytes memory lsp4MetadataURI_
+    )
+        LSP8IdentifiableDigitalAsset(
+            name_,
+            symbol_,
+            newOwner_,
+            lsp4TokenType_,
+            lsp8TokenIdFormat_
+        )
+    {
+        // set the lsp4MetadataURI
+        _setData(_LSP4_METADATA_KEY, lsp4MetadataURI_);
+    }
+}
+```
+
+#### Mint function
+
+Now let's implement the `mint(...)` function. This function will only be callable by the owner of the contract and will be in charge of deploying the LSP7 tokens as well as minting the LSP8 tokenId that will represent the LSP7 token.
+
+```typescript
+contract LSP8Token is LSP8IdentifiableDigitalAsset {
+
+  // previous code
+
+    function mint(
+        string memory nameOfLSP7_,
+        string memory symbolOfLSP7_,
+        uint256 lsp4TokenType_,
+        bool isNonDivisible_,
+        uint256 totalSupplyOfLSP7_,
+        address receiverOfInitialTokens_,
+        bytes memory lsp4MetadataURIOfLSP7_
+    ) public returns (address lsp7TokenAddress) {
```

Comment:
missing the `onlyOwner` modifier here.

```suggestion
    ) public onlyOwner returns (address lsp7TokenAddress) {
```

Reply:
I think you could also return directly the `LSP7Token` contract type? In the ABI for dApps consuming or using the tutorial, it will be converted as an `address` anyway.

```suggestion
    ) public returns (LSP7Token lsp7Token) {
```

Reply:
hmmm